manifests/params.pp

```diff
@@ -283,7 +283,7 @@
   $dhcp_managed            = true
   $dhcp_provider           = 'isc'
   $dhcp_subnets            = []
-  $dhcp_interface          = 'eth0'
+  $dhcp_interface          = pick($facts['networking']['primary'], 'eth0')
```

Comment:
On Puppet 4 with Debian AIO you don't have networking facts so this may be an issue (```Error: Evaluation Error: Operator '[]' is not applicable to an Undef Value.```). Since we're on stdlib 4.19.0 we can rely on ```fact('networking.primary')``` instead which returns undef but doesn't raise an error on undefined index thus solving the problem.